### PR TITLE
automation: feature additional headers

### PR DESCRIPTION
**Pull request by njmulsqb**
I have worked on requestor job and included the functionality to be able to add custom headers to the requests which fixes zaproxy/zaproxy#6917.

Signed-off-by: njmulsqb <najamulsaqib@tutamail.com>

This PR is opened because of accidental closure of #4087

### CODE REVIEW

**Comment by thc202** [comment]
No worries, the work is still there. That was because of the amended merge commit, reset the branch into the commit 616add5 and then soft reset to upstream/main.

**Comment by thc202** [comment]
Is your local branch already at 616add5?

**Comment by kingthorin** [comment]
```
Task :addOns:automation:test

RequestorJobUnitTest > shouldAcceptEmptyHeadersList() FAILED
    while scanning a simple key
     in 'string', line 3, column 3:
          'headers:'
          ^
    could not find expected ':'
     in 'string', line 3, column 13:
          'headers:'
                    ^
        at app//org.yaml.snakeyaml.scanner.ScannerImpl.removePossibleSimpleKey(ScannerImpl.java:533)
        at app//org.yaml.snakeyaml.scanner.ScannerImpl.fetchStreamEnd(ScannerImpl.java:607)
        at app//org.yaml.snakeyaml.scanner.ScannerImpl.fetchMoreTokens(ScannerImpl.java:320)
        at app//org.yaml.snakeyaml.scanner.ScannerImpl.checkToken(ScannerImpl.java:248)
        at app//org.yaml.snakeyaml.parser.ParserImpl$ParseBlockMappingKey.produce(ParserImpl.java:602)
        at app//org.yaml.snakeyaml.parser.ParserImpl.peekEvent(ParserImpl.java:165)
        at app//org.yaml.snakeyaml.comments.CommentEventsCollector$1.peek(CommentEventsCollector.java:59)
        at app//org.yaml.snakeyaml.comments.CommentEventsCollector$1.peek(CommentEventsCollector.java:45)
        at app//org.yaml.snakeyaml.comments.CommentEventsCollector.collectEvents(CommentEventsCollector.java:140)
        at app//org.yaml.snakeyaml.comments.CommentEventsCollector.collectEvents(CommentEventsCollector.java:119)
        at app//org.yaml.snakeyaml.composer.Composer.composeScalarNode(Composer.java:221)
        at app//org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:191)
        at app//org.yaml.snakeyaml.composer.Composer.composeValueNode(Composer.java:313)
        at app//org.yaml.snakeyaml.composer.Composer.composeMappingChildren(Composer.java:304)
        at app//org.yaml.snakeyaml.composer.Composer.composeMappingNode(Composer.java:288)
        at app//org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:195)
        at app//org.yaml.snakeyaml.composer.Composer.composeSequenceNode(Composer.java:251)
        at app//org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:193)
        at app//org.yaml.snakeyaml.composer.Composer.composeValueNode(Composer.java:313)
        at app//org.yaml.snakeyaml.composer.Composer.composeMappingChildren(Composer.java:304)
        at app//org.yaml.snakeyaml.composer.Composer.composeMappingNode(Composer.java:288)
        at app//org.yaml.snakeyaml.composer.Composer.composeNode(Composer.java:195)
        at app//org.yaml.snakeyaml.composer.Composer.getNode(Composer.java:115)
        at app//org.yaml.snakeyaml.composer.Composer.getSingleNode(Composer.java:146)
        at app//org.yaml.snakeyaml.constructor.BaseConstructor.getSingleData(BaseConstructor.java:151)
        at app//org.yaml.snakeyaml.Yaml.loadFromReader(Yaml.java:490)
        at app//org.yaml.snakeyaml.Yaml.load(Yaml.java:416)
        at app//org.zaproxy.addon.automation.jobs.RequestorJobUnitTest.shouldAcceptEmptyHeadersList(RequestorJobUnitTest.java:446)
```

**Comment by njmulsqb** [comment]
I believe the work is done from my side, please review.

**Comment by njmulsqb** [comment]
Some improvements as suggested by @thc202 via IRC.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Comment by thc202** [comment]
Actually, it's missing the changelog entry :)

File: addOns/automation/src/main/resources/org/zaproxy/addon/automation/resources/requestor-max.yaml [hunk 1]
```diff
@@ -6,4 +6,6 @@
         name:                          # String: Optional name for the request, for documentation only
         method:                        # String: A non-empty request method, default: GET
         data:                          # String: Optional data to send in the request body, supports vars
-        responseCode:                  # Int: An optional, expected response code against which the actual response code will be matched
\ No newline at end of file
+        responseCode:                  # Int: An optional, expected response code against which the actual response code will be matched
+        headers:                       # An optional list of headers
+            # - "header1:value1"
```
Comment (kingthorin) [inline addOns/automation/src/main/resources/org/zaproxy/addon/automation/resources/requestor-max.yaml:11]: Wouldn’t it make more sense for this to be between method and data?

File: addOns/automation/src/test/resources/org/zaproxy/addon/automation/resources/template-max.yaml [hunk 2]
```diff
@@ -64,6 +64,8 @@ jobs:
         method:                        # String: A non-empty request method, default: GET
         data:                          # String: Optional data to send in the request body, supports vars
         responseCode:                  # Int: An optional, expected response code against which the actual response code will be matched
+        headers:                       # An optional list of headers
+            # - "header1:value1"
```
Comment (kingthorin) [inline addOns/automation/src/test/resources/org/zaproxy/addon/automation/resources/template-max.yaml:68]: Same hwre

File: addOns/automation/src/main/resources/org/zaproxy/addon/automation/resources/requestor-max.yaml (hunk 1 above)
Comment (njmulsqb) [reply addOns/automation/src/main/resources/org/zaproxy/addon/automation/resources/requestor-max.yaml:11]: Sure, making the changes.

Comment (kingthorin) [reply addOns/automation/src/main/resources/org/zaproxy/addon/automation/resources/requestor-max.yaml]: I’m not sure how/where the plan generation is but that may also need to be re-ordered

Comment (njmulsqb) [reply addOns/automation/src/main/resources/org/zaproxy/addon/automation/resources/requestor-max.yaml]: Ok

**Review by thc202 — APPROVED** [review]

**Review by kingthorin — APPROVED** [review]
Thanks

**Pull request merged**